src/main/java/org/elasql/storage/tx/concurrency/fifolocker/FifoLockers.java

```diff
@@ -0,0 +1,202 @@
+package org.elasql.storage.tx.concurrency.fifolocker;
+
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.elasql.sql.PrimaryKey;
+
+/**
+ * FifoLockers is used to keep which transaction possesses right to access a
+ * record, and to keep those transactions which are waiting for the record.
+ * 
+ * @author Pin-Yu Wang
+ *
+ */
+public class FifoLockers {
+	private ConcurrentLinkedDeque<FifoLock> requestQueue = new ConcurrentLinkedDeque<FifoLock>();
+	private ConcurrentLinkedDeque<Long> sLockers = new ConcurrentLinkedDeque<Long>();
+	private AtomicLong xLocker = new AtomicLong(-1);
+	private PrimaryKey key;
+
+	public FifoLockers(PrimaryKey key) {
+		this.key = key;
+	}
+
+	private boolean sLocked() {
+		// avoid using sLockers.size() due to the cost of traversing the queue
+		return !sLockers.isEmpty();
+	}
```

Comment:
Suggest to remove this unused code.

Reply:
Not this one though XD, the code has been removed.